pandera/engines/pyspark_engine.py

```diff
@@ -341,10 +342,15 @@ class Date(DataType, dtypes.Date):  # type: ignore
 # timestamp
 ###############################################################################
 
+# Default timestamp equivalents
```

Comment:
If packaging is part of the dependencies for Pandera, you might consider parsing the package string to make this process more robust. Maybe not worth adding the dependency if not

```
# Default timestamp equivalents
equivalents = ["datetime", "timestamp", "TimestampType", "TimestampType()", pst.TimestampType(), pst.TimestampType]  # type: ignore

# Parse the current pyspark version.
current_version = version.parse(pyspark.__version__)

# Include new Spark 3.4 TimestampNTZType as equivalents if the current version is 3.4 or later.
if current_version >= version.parse("3.4"):
    equivalents += ["TimestampNTZType", "TimestampNTZType()", pst.TimestampNTZType, pst.TimestampNTZType()]  # type: ignore
```

Reply:
Good catch! Yes, it it:
![image](https://github.com/unionai-oss/pandera/assets/110418479/ec780fff-1e1e-454c-afbe-e392befd45cf)

Change accepted, tks!